src/base.py

```diff
@@ -152,11 +147,213 @@ def __init__(self, **kwargs):
             raise ValueError(f"dim = {self.dim} not supported")
         
         # Compute the bounding box indices for boundary conditions
-        self.boundingBoxIndices = self.bounding_box_indices()
+        self.boundingBoxIndices= self.bounding_box_indices()
         # Create boundary data for the simulation
         self._create_boundary_data()
         self.force = self.get_force()
 
+    @property
+    def lattice(self):
+        return self._lattice
+
+    @lattice.setter
+    def lattice(self, value):
+        if value is None:
+            raise ValueError("Lattice type must be provided.")
+        if self.nz == 0 and not isinstance(value, LatticeD2Q9):
+            raise ValueError("For 2D simulations, lattice type must be LatticeD2Q9.")
+        if self.nz != 0 and isinstance(self, src.models.KBCSim) and not isinstance(value, LatticeD3Q27):
+            raise ValueError("For 3D KBC simulations, lattice type must be LatticeD3Q19,")
```

Comment:
The message is incorrect. The lattice myst be D3Q27. Also I would not include this check that relyes on model specific information (KBCSim here) inside the base class. You should raise an error inside models.py for KBCSim